pkg/stub/replset.go

```diff
@@ -132,6 +133,78 @@ func (h *Handler) handleReplsetInit(m *v1alpha1.PerconaServerMongoDB, replset *v
 	return ErrNoRunningMongodContainers
 }
 
+func (h *Handler) handleStatefulSetUpdate(m *v1alpha1.PerconaServerMongoDB, set *appsv1.StatefulSet, replset *v1alpha1.ReplsetSpec, resources *corev1.ResourceRequirements) error {
+	var doUpdate bool
+
+	// Ensure the stateful set size is the same as the spec
+	err := h.client.Get(set)
+	if err != nil {
+		return fmt.Errorf("failed to get stateful set for replset %s: %v", replset.Name, err)
+	}
+	if *set.Spec.Replicas != replset.Size {
+		logrus.Infof("setting replicas to %d for replset: %s", replset.Size, replset.Name)
+		set.Spec.Replicas = &replset.Size
+		doUpdate = true
+	}
+
+	// Ensure the PSMDB version is the same as the spec
+	expectedImage := getPSMDBDockerImageName(m)
+	mongod := &set.Spec.Template.Spec.Containers[0]
+	if mongod != nil && mongod.Image != expectedImage {
+		logrus.Infof("updating spec image for replset %s: %s -> %s", replset.Name, mongod.Image, expectedImage)
+		mongod.Image = expectedImage
+		doUpdate = true
+	}
+
+	// Ensure the stateful set resources are the same as the spec
+	mongodLimits := corev1.ResourceList{
+		corev1.ResourceStorage: mongod.Resources.Limits[corev1.ResourceStorage],
+	}
+	mongodRequests := corev1.ResourceList{}
+	for _, resourceName := range []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory} {
+		mongodRequest := mongod.Resources.Requests[resourceName]
+		specRequest := resources.Requests[resourceName]
+		if specRequest.Cmp(mongod.Resources.Requests[resourceName]) != 0 {
+			logrus.Infof("updating %s resource request: %s -> %s", resourceName, mongodRequest.String(), specRequest.String())
```

Comment:
There is no need in explicit `String()` invocation here. Logrus uses `fmt` which in turn aware of Stringer interface.

Reply:
Removing .String() made the log formatting for resources change to something hard to read, example:

```
{{%!s(int64=301) %!s(resource.Scale=-3)} {%!s(*inf.Dec=<nil>)} 301m DecimalSI}
```

Where before "301m" would be printed. I will leave the .String() here until this is resolved later.